src/pages/fieldManagement/settings/DefaultFieldTable.jsx

```diff
@@ -114,30 +114,57 @@ export default function DefaultFieldTable({
             setFormSettings(getInitialFormState(siteSettings));
             onCloseEditor();
           }}
-          onSubmit={() => {
+          onSubmit={async () => {
             if (editField?.id === 'region') {
-              putSiteSetting(
+              const success = await putSiteSetting(
                 editField.backendPath,
                 formSettings.regions,
-              ).then(success => {
-                if (success) onCloseEditor();
-              });
+              );
+              if (success) onCloseEditor();
             }
             if (editField?.id === 'species') {
-              putSiteSetting(
+              const success = await putSiteSetting(
                 editField.backendPath,
                 formSettings.species,
-              ).then(success => {
-                if (success) onCloseEditor();
-              });
+              );
+              if (success) onCloseEditor();
             }
             if (editField?.id === 'relationship') {
-              putSiteSetting(
+              const newSettings = reduce(
+                formSettings.relationships,
+                (memo, currentRelationships, key) => {
+                  const dedupedRelationships = currentRelationships.filter(
+                    (entry, index) => {
+                      return (
+                        currentRelationships.indexOf(entry) === index
+                      );
+                    },
+                  );
+                  if (
+                    currentRelationships.length >
+                    dedupedRelationships.length
+                  ) {
+                    alert(
+                      intl.formatMessage({
+                        id:
+                          'DUPLICATE_RELATIONSHIPS_FOUND_AND_REMOVED',
+                      }),
+                    );
+                  }
+                  let newRelationshipChunk = {};
+                  newRelationshipChunk[key] = dedupedRelationships;
+                  return { ...memo, ...newRelationshipChunk };
+                },
+                {},
+              );
+              let newSiteSettings = {};
+              newSiteSettings['relationships'] = { ...newSettings };
+              formSettings.relationships = newSiteSettings; // I know that this is bad practice, but just passing newSiteSettings below did not seem to do the trick, and I don't know why
+              const success = await putSiteSetting(
                 editField.backendPath,
                 formSettings.relationships,
```

Comment:
this is where I tried to use newSiteSettings instead.

Reply:
i'd like to figure this out before merge - no reason to leave it dirty like this. what happens if you do 

```
const newSiteSettings = { 
    relationships: { ...newSettings }
}
const success = await putSiteSetting(editField.backendPath, newSiteSettings);
```

Reply:
Yeah... same issue that I was seeing earlier: it does not replace with the de-duplicated version! So weird!

Reply:
mmk, i will take a look at this at some point post-merge too

Reply:
In your onChange in defined in the RelationshipEditor, you are just setting the form state to relationships, which explains the behavior you are seeing. You need to include the entire form state. I think that all relationships are provided to the onChange, so the following should work:
```
onChange={relationships => {
    setFormSettings({ 
        ...formSettings,
        relationships
      });
  }}

```